--- entropy-0.4.1.11/Setup.hs.orig
+++ entropy-0.4.1.11/Setup.hs
@@ -20,7 +20,7 @@
                                         let mConf  = lookupProgram ghcProgram (withPrograms lbi)
                                             err    = error "Could not determine C compiler"
                                             cc     = locationPath . programLocation  . maybe err id $ mConf
-                                        lbiNew <- checkRDRAND cc lbi >>= checkGetrandom cc >>= checkGetentropy cc
+                                        lbiNew <- return $ addOptions ["-DHAVE_GETRANDOM", "-DHAVE_GETENTROPY"] ["-DHAVE_GETRANDOM", "-DHAVE_GETENTROPY"] lbi
                                         buildHook simpleUserHooks pd lbiNew uh bf
                       }

