$OpenBSD$

OpenBSD patch for unix-compat-mktemp.patch.

--- unix-compat-0.7.4/cbits/mktemp.c.orig
+++ unix-compat-0.7.4/cbits/mktemp.c
@@ -42,7 +42,8 @@
 #include <string.h>
 #include <unistd.h>

-#if defined(__wasm__)
+#if defined(__OpenBSD__)
+#elif defined(__wasm__)
 #include <sys/random.h>
 #else
 #include <windows.h>
@@ -162,7 +163,7 @@
     /*NOTREACHED*/
 }

-#if defined(__wasm__)
+#if defined(__OpenBSD__) || defined(__wasm__)
 static int unixcompat_random(uint32_t *value)
 {
     int r = getentropy(value, sizeof(uint32_t));
