$OpenBSD$

OpenBSD patch for entropy-getrandom.patch.

--- entropy-0.4.1.11/cbits/getrandom.c
+++ entropy-0.4.1.11/cbits/getrandom.c
@@ -1,10 +1,37 @@
-#ifdef HAVE_GETRANDOM
+#if defined(__OpenBSD__)
 
+#include <unistd.h>
+
+int system_has_getrandom()
+{
+    return 1;
+}
+
+// Returns 0 on success, non-zero on failure.
+int entropy_getrandom(unsigned char* buf, size_t len)
+{
+    while (len) {
+        size_t chunk = len > 256 ? 256 : len;
+
+        if (getentropy(buf, chunk) != 0)
+            return -1;
+
+        len -= chunk;
+        buf += chunk;
+    }
+
+    return 0;
+}
+
+#elif defined(HAVE_GETRANDOM)
+
+#if defined(HAVE_LIBC_GETRANDOM)
+
 #define _GNU_SOURCE
 #include <errno.h>
 
-#ifdef HAVE_LIBC_GETRANDOM
 #include <sys/random.h>
+
 #else
 
 #include <unistd.h>
